### PR TITLE
Merge the two goroutines in realImageGCManager#Start

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Currently two goroutines are started in realImageGCManager#Start .

This PR merges them into one goroutine.

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @Random-Liu

File: pkg/kubelet/images/image_gc_manager.go [hunk 1]
```diff
@@ -171,30 +171,30 @@ func NewImageGCManager(runtime container.Runtime, statsProvider StatsProvider, r
...
-	go wait.Until(func() {
+		counter := 0
 		images, err := im.runtime.ListImages()
 		if err != nil {
 			klog.Warningf("[imageGCManager] Failed to update image list: %v", err)
 		} else {
 			im.imageCache.set(images)
 		}
-	}, 30*time.Second, wait.NeverStop)
+		counter++
```
Comment (mattjmcnaughton) [inline pkg/kubelet/images/image_gc_manager.go]: Nit: I wonder, is there any downside to running the `detectImages` component every 30 seconds? I'm slightly worried about the impacts of using the counter on readability.

If we do decide to keep the `counter%10`, can we add a comment (or better yet, descriptive variable/func name) indicating that `counter%10` equates to once every 5 minutes?

**Review by mattjmcnaughton — APPROVED** [review]
I left a small nit that I'm happy to discuss if you would like, but overall this looks good :)

/lgtm

File: pkg/kubelet/images/image_gc_manager.go (hunk 1 above)
Comment (tedyu) [reply pkg/kubelet/images/image_gc_manager.go]:
```
any downside to running the detectImages component every 30 seconds
```
I conformed to the original period. I think we should be consistent across refactoring.

I found that I lost the branch where this PS was based on.

Please let me know whether I should open new PR for adding the comment, or the current PR is good to go - it is pretty obvious what the actual period is , dividing 10 by 2.

Thanks

File: pkg/kubelet/images/image_gc_manager.go [hunk 2]
```diff
@@ -171,30 +171,30 @@ func NewImageGCManager(runtime container.Runtime, statsProvider StatsProvider, r
...
 
+		if counter%10 == 0 {
```
Comment (dashpole) [inline pkg/kubelet/images/image_gc_manager.go]: won't `counter` always be `1` on this line?  `counter` is defined as 0 at the top of the function, and then incremented.

**Review by dashpole — CHANGES_REQUESTED** [review]
I personally think this change negatively impacts readability, and i'm not sure the elimination of a goroutine is worth it.  But @Random-Liu will have more context on what he was hoping to accomplish with the TODO.

File: pkg/kubelet/images/image_gc_manager.go (hunk 2 above)
Comment (tedyu) [reply pkg/kubelet/images/image_gc_manager.go]: counter should be defined outside the goroutine

Once @Random-Liu confirms the refactor, I will update the PR.

**Pull request closed without merging**